components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,598 @@
+      get_field_out("aero_ssa_sw_mam4").get_view<Real ***>();
+  const auto aero_tau_sw =
+      get_field_out("aero_tau_sw_mam4").get_view<Real ***>();
+
+  const auto aero_tau_lw = get_field_out("aero_tau_lw").get_view<Real ***>();
+
+  const auto aero_g_sw_eamx = get_field_out("aero_g_sw").get_view<Real ***>();
+  const auto aero_ssa_sw_eamx =
+      get_field_out("aero_ssa_sw").get_view<Real ***>();
+  const auto aero_tau_sw_eamx =
+      get_field_out("aero_tau_sw").get_view<Real ***>();
+
+  const auto aero_tau_forward =
+      get_field_out("aero_tau_forward").get_view<Real ***>();
 
-void MAMOptics::finalize_impl()
-{
+  const auto aodvis = get_field_out("aodvis").get_view<Real *>();
+
+  // Compute optical properties on all local columns.
+  // (Strictly speaking, we don't need this parallel_for here yet, but we
+  // leave
+  //  it in anticipation of column-specific aerosol optics to come.)
+
+  // NOTE! we need a const mam_coupling::DryAtmosphere dry_atm for gpu access.
+  // We cannot use member of this class inside of the parallel_for
+  const mam_coupling::DryAtmosphere &dry_atm = dry_atm_;
+  const auto &p_int                          = p_int_;
+  const auto &p_del                          = p_del_;
+  const auto &ssa_cmip6_sw                   = ssa_cmip6_sw_;
+  const auto &af_cmip6_sw                    = af_cmip6_sw_;
+  const auto &ext_cmip6_sw                   = ext_cmip6_sw_;
+  const auto &ext_cmip6_lw                   = ext_cmip6_lw_;
+  const auto &work                           = work_;
+  const auto &dry_aero                       = dry_aero_;
+  const auto &aerosol_optics_device_data     = aerosol_optics_device_data_;
+  // const auto& z_int =z_iface;
+  Kokkos::parallel_for(
+      policy, KOKKOS_LAMBDA(const ThreadTeam &team) {
+        const Int icol     = team.league_rank();  // column index
+        auto odap_aer_icol = ekat::subview(aero_tau_lw, icol);
+        const auto atm     = mam_coupling::atmosphere_for_column(dry_atm, icol);
+
+        // FIXME: Get rid of this
+        // auto nccn = ekat::subview(aero_nccn, icol);
+
+        // FIXME: interface pressure [Pa]
+        auto pint = ekat::subview(p_int, icol);
+        // FIXME: dry mass pressure interval [Pa]
+        // FIXME:
+        auto zi      = ekat::subview(dry_atm.z_iface, icol);
+        auto pdel    = ekat::subview(p_del, icol);
+        auto pdeldry = ekat::subview(dry_atm.p_del, icol);
+
+        auto ssa_cmip6_sw_icol = ekat::subview(ssa_cmip6_sw, icol);
+        auto af_cmip6_sw_icol  = ekat::subview(af_cmip6_sw, icol);
+        auto ext_cmip6_sw_icol = ekat::subview(ext_cmip6_sw, icol);
+        auto ext_cmip6_lw_icol = ekat::subview(ext_cmip6_lw, icol);
+
+        // FIXME: check if this correct: Note that these variables have pver+1
+        // levels tau_w =>  aero_ssa_sw  (pcols,0:pver,nswbands) ! aerosol
+        // single scattering albedo * tau
+        auto tau_w_icol = ekat::subview(aero_ssa_sw, icol);
```

Comment:
I'm a little confused here. You are defining these product quantities to be subviews of the actual optical property views. I.e., tau_w should be the product tau * w (or tau * ssa in the convention used for the views). To get what I would expect from something named aero_ssa_sw, I would think you would need to do aero_ssa_sw = tau_w / tau at some point. Is this the case? It looks like these subviews are passed to the aer_rad_props routines below, which I assume produce the products, not the raw optical properties.

Reply:
You are correct. the views are passed to  `aer_rad_props::aer_rad_props_sw ` to compute `tau_w`; [see here](https://github.com/eagles-project/mam4xx/blob/990c83c1d237d7a6aa61640976500d6fed7c024a/src/mam4xx/modal_aer_opt.hpp#L781). We are missing this operations: `aero_ssa_sw = tau_w / tau `.

Reply:
These operations were corrected.